File: leetcode/C/9. Plaindrome Number/main.c
#include <stdint.h>
#include <stdbool.h>

bool isPalindrome(int number) {
    int digit;
    uint32_t tmp = number, reversed_number = 0;

    if (number < 0)
        return false;

    if (number == 0)
        return true;
  
    while (tmp > 0) {
        digit = tmp % 10;
        tmp /= 10;
        reversed_number = (reversed_number * 10) + digit;
    }

    return number == reversed_number;
}

File: leetcode/Python/9. Plaindrome Number/main.py
class Solution:
    def isPalindrome(self, number: int) -> bool:
        if number < 0:
            return False
        if number == 0:
            return True

        tmp = number
        reversed_number = 0

        while tmp > 0:
            digit = tmp % 10
            tmp //= 10
            reversed_number = reversed_number * 10 + digit

        return number == reversed_number
